### PR TITLE
Collective dispatching from Process Group

### PR DESCRIPTION
**Pull request by H-Huang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #91257

Fixes https://github.com/pytorch/pytorch/issues/90932
Fixes https://github.com/pytorch/pytorch/issues/90659

Remove redundant collection operation definitions by calling the ops directly from `ProcessGroup`

Context:
https://github.com/pytorch/pytorch/issues/86225

Differential Revision: [D42854676](https://our.internmc.facebook.com/intern/diff/D42854676)

### CODE REVIEW

**Comment by H-Huang** [comment]
@H-Huang has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D42854676).

File: torch/csrc/distributed/c10d/Ops.cpp [hunk 1]
```diff
@@ -58,387 +58,6 @@ TORCH_LIBRARY(c10d, m) {
...
-    int64_t tag) {
-  static auto op = c10::Dispatcher::singleton()
-                       .findSchemaOrThrow("c10d::recv_any_source_", "")
-                       .typed<c10::intrusive_ptr<::c10d::Work>(
-                           at::TensorList,
-                           const c10::intrusive_ptr<::c10d::ProcessGroup>&,
-                           int64_t)>();
-  return op.call(tensors, process_group, tag);
-}
-
```
Comment (kwen2501) [inline torch/csrc/distributed/c10d/Ops.cpp:441]: After removing this code block, should we also remove the corresponding API declarations in Ops.hpp?

**Review by kwen2501 — APPROVED** [review]
Thanks for the clean-up! LGTM.
Please see my inline comment.

File: torch/csrc/distributed/c10d/Ops.cpp (hunk 1 above)
Comment (H-Huang) [reply torch/csrc/distributed/c10d/Ops.cpp:441]: Good point! Thanks

**Comment by H-Huang** [comment]
@H-Huang has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D42854676).

**Pull request closed without merging**